core/templates/dev/head/pages/admin/misc_tab/AdminMiscTabDirective.js

```diff
@@ -76,6 +79,54 @@ oppia.directive('adminMiscTab', [
         $scope.downloadTopicSimilaritiesFile = function() {
           $window.location.href = ADMIN_TOPICS_CSV_DOWNLOAD_HANDLER_URL;
         };
+
+        var setDataExtractionQueryStatusMessage = function(message) {
+          $scope.showDataExtractionQueryStatus = true;
+          $scope.dataExtractionQueryStatusMessage = message;
+        };
+
+        $scope.submitQuery = function() {
+          var STATUS_PENDING = (
+            'Data extraction query has been submitted. Please wait.');
+          var STATUS_FINISHED = 'Loading the extracted data ...';
+          var STATUS_FAILED = 'Error, ';
+
+          var data = {
+            exp_id: $scope.expId,
+            exp_version: $scope.expVersion,
+            state_name: $scope.stateName,
+            num_answers: $scope.numAnswers
+          };
+
+          setDataExtractionQueryStatusMessage(STATUS_PENDING);
+
+          $http.post(
+            DATA_EXTRACTION_QUERY_HANDLER_URL, data
+          ).then(
+            function(response) {
+              var data = response.data;
+              setDataExtractionQueryStatusMessage(STATUS_FINISHED);
+              var extractedAnswers = JSON.stringify(data.data);
+              var dataURL = (
+                'data:application/json;base64,' + btoa(extractedAnswers));
+              var downloadURL = (
+                '<a href="' + dataURL +
```

Comment:
That was my original plan but exploration history uses GET url to download that data, so browser can download whatever it gets from url. All we have to do is just `window.open()`. That is not the case here, where POST url is used. Browser can't access POST url using `window.open()`.

We are using POST because we are not sure whether data can be obtained or not (like exp does not exists or state_name is wrong). Because of this uncertainty I decided to use POST so we can state the error in user's query.

One alternative is to have a download button (href) so that it can be downloaded to browser. This way we can avoid generating html on the go but we will have to use data-url hyperlink  so that content can be downloaded. What do you suggest?